SteamKit2/SteamKit2/Steam/SteamClient/SteamClient.cs

```diff
@@ -172,12 +174,7 @@ public void RemoveHandler( ClientMsgHandler handler )
         {
             Type type = typeof( T );
 
-            if ( handlers.Contains( type ) )
-            {
-                return handlers[ type ] as T;
-            }
-
-            return null;
+            return handlers.Find( h => h.GetType() == type ) as T;
```

Comment:
The `GetHandler` call is the only place where it realistically suffers, but I don't see why you need to be calling this method a lot.